opm/simulators/wells/VFPProdProperties.cpp

```diff
@@ -215,7 +215,7 @@ calculateBhpWithTHPTarget(const std::vector<double>& ipr_a,
 
     std::vector<detail::RateBhpPair> ratebhp_samples;
     for (size_t i = 0; i < flo_samples.size(); ++i) {
-        ratebhp_samples.push_back( detail::RateBhpPair{flo_samples[i], bhp_flo_samples[i]} );
+        ratebhp_samples.push_back( detail::RateBhpPair{-flo_samples[i], bhp_flo_samples[i]} );
     }
```

Comment:
@atgeirr Can you double check this change? I think the current code is wrong, but I may have miss something.

Reply:
Not sure if I see it correctly.  It does not looks like this function is used anywhere now.

Reply:
If I read correctly, with the iterative solver, this intersection algorithm is not used anymore. With the iterative solver, the good thing is that the IPR is not used directly. And accurate IPR was a problem there.

Reply:
> Not sure if I see it correctly. It does not looks like this function calculateBhpWithTHPTarget is used anywhere now.¨

That may explain why what looks like a wrong sign could have survived in the code for some time, and explains why I didn't notice any affect of changing this.

Reply:
Referring to the comment above,
```
    // kind of unncessarily following the tradation that producers should have negative rates
    // the key is here that it should be consistent with the function bhpwithflo
```

Not sure it is wrong here, or whether it matters here. It is a very old code and it must succeed with some VFP/THP control cases before. I will hesitate to say the sign is necessarily wrong or whether it matters, or other later changes cause inconsistency.  If it were wrong from the very beginning, it would cause big issues much earlier.  But I might oversee something and be wrong here.

Reply:
This is how I read the code. 

When the production rate is computed it is positive
Then

```
   // kind of unncessarily following the tradation that producers should have negative rates
    // the key is here that it should be consistent with the function bhpwithflo
    for (double& value : flo_samples) {
        value = -value;
    }

```

which makes it negative since bhpwithflo expects negative production rates. 

Then it is stored as a pair in `ratebhp_samles`, which again is used in `findIntersectionForBhp`. 

But as I read it. `findIntersectionForBhp ` expects positive signed rates

@GitPaean I would very much like you to go through the same code and check if you agree with my understanding. The constant flipping of signs may have "flipped" my brain as well.

Reply:
I am not saying you are wrong, excuse me if you feel that way. 
I was trying to put a breakpoint to check how things work there, then it never got reached. Then I saw it is not used anymore. 

Your reading is probably correct.  With a very quick go through, can you tell me why `findIntersectionForBhp` needs positive signed rates?

Reply:
> I am not saying you are wrong, excuse me if you feel that way.

Not at all. I really appreciate your eyes on this. I am not 100% sure myself I would not like this to be merged before one more developer has looked carefully at this. 


> then it never got reached.

If this is stale code. I suggest we just delete it. 

The signs of the fluxes in `findIntersectionForBhp ` needs to be consistent. flo_bhp_limit and flo_bhp_middle is positive I think i.e. flow_samles should be the same.